### ISSUE TITLE
Compute V2: Image reference is hidden Server JSON serialization

### ISSUE THREAD

**Issue by artpej** [gophercloud/gophercloud#1618]
The `Image` reference is hidden for the `Server` JSON serialization:

https://github.com/gophercloud/gophercloud/blob/46d131b669762bc8c44f6523abd7c8a8e1df1a8b/openstack/compute/v2/servers/results.go#L184

it could be usefull according I didn't found another way to get the image reference of a running server.

**Comment by jtopjian** [comment]
The image reference is built here:

https://github.com/gophercloud/gophercloud/blob/46d131b669762bc8c44f6523abd7c8a8e1df1a8b/openstack/compute/v2/servers/results.go#L239-L247

I think in newer microversions, the image structure has changed. Try to make sure you are using API version 2.1 without a microversion and see if that helps. If it does, we'll have to look into amending the server result for a newer image format.

**Comment by artpej** [comment]
Unmarshalling works fine for me, my use case is to serve directly the results through a webservice by re-serialize them in JSON but the custom Mashaller is not present.

**Comment by jtopjian** [comment]
You should be able to do something like:

```go
type MyImage struct {
  Image map[string]interface{} `json:"image"`
}

var onlyImage MyImage

res := servers.Get(client, "id")
err := res.ExtractInto(&onlyImage)
```

**Comment by artpej** [comment]
Yes, `ExtractInto` and `̀ExtractServersInto` saved me today :wink:.

Is there any plan to also provide Marshaling of result resources? 

Whatever, many thanks @jtopjian for your help.

**Comment by jtopjian** [comment]
Not at this time. If you could explain a detailed use-case of why it would be useful, we can leave it here as a feature request.

**Comment by apuzyrevsky** [comment]
Hi @jtopjian ! I've encountered on the same issue as well and I am wondering why did you set an image map to "-" instruct, cause when I changed it to the "image", it works fine. And is there an option to serve service results through the Allpages method with image info as well?
